src/cata_tiles.cpp

```diff
@@ -533,34 +555,17 @@ void cata_tiles::draw(int destx, int desty, int centerx, int centery, int width,
     screentile_width = (width + tile_width - 1) / tile_width;
     screentile_height = (height + tile_height - 1) / tile_height;
 
-    for (int my = 0; my < sy; ++my) {
-        for (int mx = 0; mx < sx; ++mx) {
-            x = mx + o_x;
-            y = my + o_y;
-            l = light_at(x, y);
-            const auto critter = g->critter_at( x, y );
-            if (l != CLEAR) {
-                // Draw lighting
-                draw_lighting(x, y, l);
-                if( critter != nullptr && g->u.sees_with_infrared( *critter ) ) {
-                    draw_from_id_string( "infrared_creature", C_NONE, empty_string, x, y, 0, 0 );
-                }
-                continue;
-            }
-            // light is no longer being considered, for now.
-            // Draw Terrain if possible. If not possible then we need to continue on to the next part of loop
-            if (!draw_terrain(x, y)) {
-                continue;
-            }
-            draw_furniture(x, y);
-            draw_trap(x, y);
-            draw_field_or_item(x, y);
-            draw_vpart(x, y);
-            if( critter != nullptr ) {
-                draw_entity( *critter, x, y );
-            }
-        }
-    }
+    auto map_draw_function = [&](int x, int y, lit_level ll) -> void {
+        draw_specific_tile(x, y, ll);
+    };
+
+    g->m.draw_loop(
```

Comment:
In some places the loop goes from 0 to N, then gets added to base coordinates inside the loop. In some places the loop goes from min to max, then gets converted to display coordinates inside the loop. I'm just not seeing the consistency necessary for the easy understanding you're claiming. And I say this as someone who has just had to go through the code in question and got confused many times.

PS: I do plan to add an isometric renderer, which would use this new system, or some patch on top of this new system.